# 다중공선성 - VIF 구하기

In [1]:
import pandas as pd

In [14]:
df = pd.read_excel("C:/Users/김나영/Desktop/data/small_used_car.xlsx")
df.columns

Index(['mileage', 'model', 'price', 'year', 'my_car_damage',
       'other_car_damage'],
      dtype='object')

In [13]:
from statsmodels.formula.api import ols

In [15]:
# price~나머지 변수의 회귀분석 모델을 만들려고 함
# 이때 이때 mileage의 VIF를 구하려고 함
res = ols('mileage ~ model+year+my_car_damage+other_car_damage', data=df).fit()

In [16]:
res.summary()

<class 'statsmodels.iolib.summary.Summary'>
"""
                            OLS Regression Results                            
==============================================================================
Dep. Variable:                mileage   R-squared:                       0.338
Model:                            OLS   Adj. R-squared:                  0.328
Method:                 Least Squares   F-statistic:                     34.36
Date:                Wed, 28 Oct 2020   Prob (F-statistic):           3.62e-23
Time:                        15:55:06   Log-Likelihood:                -3255.1
No. Observations:                 274   AIC:                             6520.
Df Residuals:                     269   BIC:                             6538.
Df Model:                           4                                         
Covariance Type:            nonrobust                                         
====================================================================================
                       coef    std err          t      P>|t|      [0.025      0.975]
------------------------------------------------------------------------------------
Intercept         1.956e+07    1.8e+06     10.861      0.000     1.6e+07    2.31e+07
model[T.K3]       -160.5713   5003.433     -0.032      0.974      -1e+04    9690.298
year             -9677.1395    894.403    -10.820      0.000   -1.14e+04   -7916.218
my_car_damage        0.0005      0.001      0.679      0.498      -0.001       0.002
other_car_damage    -0.0011      0.000     -2.319      0.021      -0.002      -0.000
==============================================================================
Omnibus:                       45.374   Durbin-Watson:                   1.670
Prob(Omnibus):                  0.000   Jarque-Bera (JB):               86.746
Skew:                           0.871   Prob(JB):                     1.46e-19
Kurtosis:                       5.137   Cond. No.                     4.56e+09
==============================================================================

Warnings:
[1] Standard Errors assume that the covariance matrix of the errors is correctly specified.
[2] The condition number is large, 4.56e+09. This might indicate that there are
strong multicollinearity or other numerical problems.
"""

### VIF 공식 = 1/(1-R^2)

In [18]:
vif = 1/(1-0.338)
vif

1.5105740181268883

5,10이상일 때 다중공선성에 문제가 있다고 보는데 여기선 1.5이므로 문제 없음!